schema-engine/sql-introspection-tests/tests/multi_schema/mysql.rs

```diff
@@ -0,0 +1,806 @@
+          id Int @id
+
+          @@map("A")
+          @@schema("first_m3")
+        }
+
+        model second_m3_A {
+          id Int @id
+
+          @@map("A")
+          @@schema("second_m3")
+        }
+    "#]];
+
+    api.expect_datamodel(&expected).await;
+
+    let expected = expect![[r#"
+        *** WARNING ***
+
+        These items were renamed due to their names being duplicates in the Prisma schema:
+          - Type: "model", name: "first_m3_A"
+          - Type: "model", name: "second_m3_A"
+    "#]];
+
+    api.expect_warnings(&expected).await;
+
+    Ok(())
+}
+
+#[test_connector(tags(Mysql), preview_features("multiSchema"), namespaces("first_m4", "second_m4"))]
+async fn multiple_schemas_w_duplicate_sanitized_table_names_are_introspected(api: &mut TestApi) -> TestResult {
+    let schema_name = "first_m4";
+    let other_name = "second_m4";
+    let setup = formatdoc! {
+        r#"
+             CREATE SCHEMA `{schema_name}`;
+             CREATE TABLE `{schema_name}`.`2A` (id INT PRIMARY KEY);
+
+             CREATE SCHEMA `{other_name}`;
+             CREATE TABLE `{other_name}`.`1A` (id INT PRIMARY KEY);
+         "#
+    };
+    api.raw_cmd(&setup).await;
+
+    let expected = expect![[r#"
+        generator client {
+          provider        = "prisma-client-js"
+          previewFeatures = ["multiSchema"]
+        }
+
+        datasource db {
+          provider = "mysql"
+          url      = "env(TEST_DATABASE_URL)"
+          schemas  = ["first_m4", "second_m4"]
+        }
+
+        model first_m4_2A {
+          id Int @id
+
+          @@map("2A")
+          @@schema("first_m4")
+        }
+
+        model second_m4_1A {
+          id Int @id
+
+          @@map("1A")
+          @@schema("second_m4")
+        }
+    "#]];
+
+    api.expect_datamodel(&expected).await;
+
+    let expected = expect![[r#"
+        *** WARNING ***
+
+        These items were renamed due to their names being duplicates in the Prisma schema:
+          - Type: "model", name: "first_m4_2A"
+          - Type: "model", name: "second_m4_1A"
+    "#]];
+
+    api.expect_warnings(&expected).await;
+
+    Ok(())
+}
+
+#[test_connector(tags(Mysql), preview_features("multiSchema"), namespaces("first_m5", "second_m5"))]
+async fn multiple_schemas_w_cross_schema_are_introspected(api: &mut TestApi) -> TestResult {
+    let schema_name = "first_m5";
+    let other_name = "second_m5";
+    let create_schema = format!("CREATE Schema `{schema_name}`",);
+    let create_table = format!("CREATE TABLE `{schema_name}`.`A` (id INT PRIMARY KEY)",);
+    //Todo
+    api.database().raw_cmd(&create_schema).await?;
+    api.database().raw_cmd(&create_table).await?;
+
+    let create_schema = format!("CREATE Schema `{other_name}`",);
+    let create_table =
+        format!("CREATE TABLE `{other_name}`.`B` (id INT PRIMARY KEY, fk INT, CONSTRAINT fk_tbl FOREIGN KEY (fk) REFERENCES `{schema_name}`.`A`(`id`) ON DELETE RESTRICT ON UPDATE RESTRICT)",);
```

Comment:
ℹ️ Difference to Postgres: Have to specify constraints with the dedicated `FOREIGN KEY` construct here and not just inline with the column name.
Also specified the `ON DELETE` and `ON UPDATE` rules explicitly to work around differences in mysql 5 vs 8. For mysql `NO ACTION` and `RESTRICTED` behave the same, but there are different defaults you get when introspecting between the mysql versions.